pico_w/udp_beacon/picow_udp_beacon.c

```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright (c) 2022 Andrew McDonnell
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include <string.h>
+#include <stdlib.h>
+
+#include "pico/stdlib.h"
+#include "pico/cyw43_arch.h"
+
+#include "lwip/pbuf.h"
+#include "lwip/udp.h"
+
+#define UDP_PORT 4444
+#define BEACON_MSG_LEN_MAX 127
+#define BEACON_TARGET "255.255.255.255"
+#define BEACON_INTERVAL_MS 1000
+
+void run_udp_beacon() {
+    struct udp_pcb* pcb = udp_new();
+
+    ip_addr_t addr;
+    ipaddr_aton(BEACON_TARGET, &addr);
+
+    int counter = 0;
+    while (true) {
+        struct pbuf *p = pbuf_alloc(PBUF_TRANSPORT, BEACON_MSG_LEN_MAX+1, PBUF_RAM);
+        char *req = (char *)p->payload;
+        memset(req, 0, BEACON_MSG_LEN_MAX+1);
+        snprintf(req, BEACON_MSG_LEN_MAX, "%d\n", counter);
+        err_t er = udp_sendto(pcb, p, &addr, UDP_PORT);
+        pbuf_free(p);
+        if (er != ERR_OK) {
+            printf("Failed to send UDP packet! error=%d", er);
+        } else {
+            printf("Sent packet %d\n", counter);
+            counter++;
+        }
+
+        // Note in practice for this simple UDP transmitter,
+        // the end result for both background and poll is the same
+
+#if PICO_CYW43_ARCH_POLL
+        // if you are using pico_cyw43_arch_poll, then you must poll periodically from your
+        // main loop (not from a timer) to check for WiFi driver or lwIP work that needs to be done.
+        cyw43_arch_poll();
+        sleep_ms(BEACON_INTERVAL_MS);
```

Comment:
Looking at this again - it's a bit of bad form to block when polling. While it works, we might not want anyone to copy this...

Reply:
I figured it would be fit for purpose for such a simple example... but I'll have a look and come up with something more advanced if that is needed

Reply:
Lets stick with what we've got for now. I can push another change to improve things later.